polygon/bridge/bridge.go

```diff
@@ -0,0 +1,200 @@
+package bridge
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ledgerwatch/log/v3"
+
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon/common/u256"
+	"github.com/ledgerwatch/erigon/core"
+	"github.com/ledgerwatch/erigon/core/state"
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/node"
+	"github.com/ledgerwatch/erigon/node/nodecfg"
+	"github.com/ledgerwatch/erigon/polygon/bor/borcfg"
+	"github.com/ledgerwatch/erigon/polygon/heimdall"
+)
+
+type fetchSyncEventsType func(ctx context.Context, fromId uint64, to time.Time, limit int) ([]*heimdall.EventRecordWithTime, error)
+
+type IDRange struct {
+	start, end uint64
+}
+
+type Bridge struct {
+	DB                       kv.RwDB
+	log                      log.Logger
+	borConfig                *borcfg.BorConfig
+	ready                    bool
+	eventMap                 map[uint64]IDRange // block num to eventID range
+	lastProcessedBlockNumber uint64
+	lastProcessedEventID     uint64
+	stateClientAddress       libcommon.Address
+
+	fetchSyncEvents fetchSyncEventsType
+}
+
+func NewBridge(ctx context.Context, config *nodecfg.Config, name string, readonly bool, logger log.Logger, borConfig *borcfg.BorConfig, fetchSyncEvents fetchSyncEventsType) (*Bridge, error) {
+	// create new db
+	db, err := node.OpenDatabase(ctx, config, kv.PolygonBridgeDB, name, readonly, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Bridge{
+		DB:                       db,
+		log:                      logger,
+		borConfig:                borConfig,
+		fetchSyncEvents:          fetchSyncEvents,
+		eventMap:                 map[uint64]IDRange{},
+		lastProcessedBlockNumber: 0,
+		lastProcessedEventID:     0,
+		stateClientAddress:       libcommon.HexToAddress(borConfig.StateReceiverContract),
+	}, nil
+}
+
+func (b *Bridge) Run(ctx context.Context) error {
```

Comment:
it will be good if you write a test where you mock heimdall (for the fetch state sync events API) -> then call bridge.Run -> bridge.ProcessNewBlocks -> bridge.GetEvents and assert you got the right events